lib/tasks/uploads.rake

```diff
@@ -1037,16 +1037,25 @@ def fix_missing_s3
       else
         # we do not fix sha, it may be wrong for arbitrary reasons, if we correct it
         # we may end up breaking posts
-        upload.assign_attributes(etag: fixed_upload.etag, url: fixed_upload.url, verification_status: Upload.verification_statuses[:unchecked])
-        upload.save!(validate: false)
+        save_error = nil
+        begin
+          upload.assign_attributes(etag: fixed_upload.etag, url: fixed_upload.url, verification_status: Upload.verification_statuses[:unchecked])
+          upload.save!(validate: false)
+        rescue => save_error
+          # url might be null
+        end
```

Comment:
```suggestion
        upload.assign_attributes(etag: fixed_upload.etag, url: fixed_upload.url, verification_status: Upload.verification_statuses[:unchecked])
        saved = upload.save(validate: false)
```

(then invert the `if` block below using the `saved` var)

Reply:
Made the change.